### PR TITLE
Lab 1 & 2

### PR DESCRIPTION
**Pull request by AndreeaPloscar**

### CODE REVIEW

File: Lexic.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		<nonzeroDigit> ::= 1|2|3|4 … |9
+		<digit_seq> ::= <digit> | <digit><digit_seq>
+		<unsignedNumber> ::= <nonzeroDigit> | <nonzeroDigit><digit_seq>
+		<int> ::= <zero> |  <sign><unsignedNumber> | <unsignedNumber>
+
+		- str
+
+		<strSequence> ::= <chr> | <chr><strSequence>
+	    	<chr> ::= <letter>|<digit>|.
+		<str> ::= "<strSequence>"
```
Comment (alex-nedelcu) [inline Lexic.txt:34]: i think the double quotes should not be present here.
only in the syntax file

File: Syntax.in [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<program> ::= #<stmtList>#
+
+<stmtList> ::= <stmt> | <stmt><stmtList>
+
+<stmt> ::= <simpleStmt> | <structStmt> 
+
+<simpleStmt> ::= <declaration> | <iostmt> | <assignstmt> 
+
+<declaration> ::= var <identifier> :: <type> | var <identifier> :: <type> = <const>
```
Comment (alex-nedelcu) [inline Syntax.in:9]: `<identifier>` must be redefined also here

File: Syntax.in [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+<const> ::= <char> | <int> | <str>
```
Comment (alex-nedelcu) [inline Syntax.in:11]: same for `<char> <int> <str>` - they must be redefined in this file as well

File: Syntax.in [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+<identifierList> ::= <identifier> | <identifier>,<identifierList>
+
+<simpleAssignOp> ::= :=
+
+<specialAssignOp> ::= +:= | -:=
+
+<simpleAssign> ::= <identifier> <simpleAssignOp> <expression> 
+
+<specialAssign> ::= <identifier> <specialAssignOp> <const>
```
Comment (alex-nedelcu) [inline Syntax.in:29]: so you cannot have something like `x +:= y`
if that's on purpose it's fine. just mentioned in case you forgot

**Review by alex-nedelcu — APPROVED** [review]
small suggestions - rest looks good

File: Syntax.in (hunk 4 above)
Comment (AndreeaPloscar) [reply Syntax.in:29]: it is on purpose

**Pull request merged**